packages/toolkit/src/view/model/view-model/ModelOverview.tsx

```diff
@@ -115,98 +126,106 @@ export const ModelOverview = ({ model, modelState }: ModelOverviewProps) => {
   const existingModelTriggerResult = useLastModelTriggerResult({
     accessToken,
     modelName: model?.name || null,
-    fullView: showFullTriggerResult,
-    enabled:
-      enabledQuery && (!existingTriggerState || existingTriggerState.done),
+    fullView: true,
+    enabled: enabledQuery,
   });
 
-  useEffect(() => {
-    if (!existingModelTriggerResult.data?.operation || !model) {
-      return;
-    }
+  const pollForResponse = React.useCallback(async () => {
+    await queryClient.invalidateQueries({
+      queryKey: ["models", "operation", model?.name],
+    });
 
+    forceUpdate();
+  }, [forceUpdate]);
```

Comment:
@orangecoloured This is a bit hacky to me, normally when the cached data changed, all the related subscribed components and hook should change too. I am wondering what is the issue here?

Reply:
I invalidate the response here and force the hooks to execute again, so the query hook actually tries to fetch the new data and not get it from cache. Basically I'm trying to mimic executing a simple fetch with a timeout.

Reply:
This is where I am a bit lost, please shed some light on it

```ts
await queryClient.invalidateQueries({
      queryKey: ["models", "operation", model?.name],
    });
```

This will invalidate the operation query. And ideally, the updated data will trigger the re-render so you can get the up-to-date data. 

Then you trigger the `forceUpdate` function which make the whole component re-render. If the react-query state change already trigger the re-render. Why you need to jump in the life-cycle and force it to render?

Reply:
Do you mean that calling `invalidateQueries` triggers the hook to fetch the data? I thought it just invalidates the cached response so next time the hook is executed it makes a new fetch instead of giving the cached result.

Reply:
> Waiting for queries to become stale before they are fetched again doesn't always work, especially when you know for a fact that a query's data is out of date because of something the user has done. For that purpose, the QueryClient has an invalidateQueries method that lets you intelligently mark queries as stale and potentially refetch them too!

https://tanstack.com/query/v4/docs/framework/react/guides/query-invalidation

According to the docs, in some scenario `invalidateQueries` will trigger refetch. But some scenario don't, it depends on whether your query is already rendered or not.

If you want a much straight, you can use refetch function (At the end of this page https://tanstack.com/query/latest/docs/framework/react/reference/useQuery)

![CleanShot 2024-06-24 at 20 22 48](https://github.com/instill-ai/console/assets/57251712/9c971d42-6ca9-473c-82ba-74e413568071)

Reply:
Fixed